tests_config/run_spec.js

```diff
@@ -15,8 +15,11 @@ const prettier = !TEST_STANDALONE
   : require("prettier/standalone");
 
 global.run_spec = (dirname, parsers, options) => {
-  // istanbul ignore next
-  if (!parsers || !parsers.length) {
+  // `isNoParserTests` mean to test `inferParser` on `standalone`
+  const isNoParserTests = dirname.includes("no-parser-tests");
+  if (isNoParserTests) {
+    parsers = [];
+  } else if (!parsers || !parsers.length) {
     throw new Error(`No parsers were specified for ${dirname}`);
```

Comment:
Is this check really needed? I think we can remove it.

Reply:
Maybe new tests forgot add parsers, I think we should keep it

Reply:
Added in https://github.com/prettier/prettier/pull/3356 , let's keep it. but `// istanbul ignore next` can be removed...

Reply:
What's wrong with allowing Prettier to infer the parser if it's what the author of the test wants? I guess this check was added when the parser inference wasn't as good as it is today.

Reply:
I guess, if someone want test 'foo.js' on `flow` but forgot add parser, the default parser `babel` will work, and a little hard to find the mistake?